src/repr/src/row/encoding.rs

```diff
@@ -156,7 +156,13 @@ impl ColumnType {
 /// `VarChar`).
 pub trait DatumToPersist {
     /// The persist columnar type we're mapping to/from.
-    type Data: Data;
+    type Data: Data<Cfg = Self::Cfg>;
+
+    // TODO: All DatumToPersist impls have a `()` config, so it would be lovely
```

Comment:
does this currently preclude us from creating a `DynStruct` for a `ScalarType::Record`?

Reply:
ah good point! I think this means we can just remove the TODO, but let me mull that over for a bit before I commit to it

Reply:
done